python/cudf_polars/docs/overview.md

```diff
@@ -198,6 +243,123 @@ To simplify state tracking, all columns should be considered immutable
 on construction. This matches the "functional" description coming from
 the logical plan in any case, so is reasonably natural.
 
+## Traversing and transforming nodes
+
+As well as just representing and evaluating nodes. We also provide
+facilities for traversing a tree of nodes and defining transformation
+rules in `dsl/traversal.py`. The simplest is `traversal`, a
+[pre-order](https://en.wikipedia.org/wiki/Tree_traversal) visit of all
+unique nodes in an expression. Use this if you want to know some
+specific thing about an expression. For example, to determine if an
+expression contains a `Literal` node:
+
+```python
+def has_literal(node: Expr) -> bool:
+    return any(isinstance(e, Literal) for e in traversal(node))
+```
+
+For transformations and rewrites, we use the following generic
+pattern. Rather than defining methods on each node in turn for a
+particular rewrite rule, we prefer free functions and use
+`functools.singledispatch` to provide dispatching.
```

Comment:
Let's remove the first sentence here. It doesn't add anything, especially given the sentence in the next paragraph that says "We therefore adopt ...". The comment about singledispatching would fit better around line 284 or so where we're discussing registering handlers.

Reply:
Agreed reworked a bit.